flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -135,154 +142,165 @@ public void testAbortOldAndStartNewCheckpoint() throws Exception {
                     worker.processAllRequests();
 
                     ChannelStateWriteResult result42 = writer.getAndRemoveWriteResult(checkpoint42);
-                    assertTrue(result42.isDone());
-                    try {
-                        result42.getInputChannelStateHandles().get();
-                        fail("The result should have failed.");
-                    } catch (Throwable throwable) {
-                        assertTrue(findThrowable(throwable, TestException.class).isPresent());
-                    }
+                    assertThat(result42.isDone()).isTrue();
+                    assertThatThrownBy(() -> result42.getInputChannelStateHandles().get())
+                            .as("The result should have failed.")
+                            .hasCauseInstanceOf(TestException.class);
 
                     ChannelStateWriteResult result43 = writer.getAndRemoveWriteResult(checkpoint43);
-                    assertFalse(result43.isDone());
+                    assertThat(result43.isDone()).isFalse();
                 });
     }
 
-    @Test(expected = TestException.class)
-    public void testBuffersRecycledOnError() throws Exception {
-        unwrappingError(
-                TestException.class,
-                () -> {
-                    NetworkBuffer buffer = getBuffer();
-                    try (ChannelStateWriterImpl writer =
-                            new ChannelStateWriterImpl(
-                                    TASK_NAME, new ConcurrentHashMap<>(), failingWorker(), 5)) {
-                        writer.open();
-                        callAddInputData(writer, buffer);
-                    } finally {
-                        assertTrue(buffer.isRecycled());
-                    }
-                });
+    @Test
+    void testBuffersRecycledOnError() {
+        assertThatThrownBy(
+                        () -> {
+                            NetworkBuffer buffer = getBuffer();
+                            try (ChannelStateWriterImpl writer =
+                                    new ChannelStateWriterImpl(
+                                            TASK_NAME,
+                                            new ConcurrentHashMap<>(),
+                                            failingWorker(),
+                                            5)) {
+                                writer.open();
+                                callAddInputData(writer, buffer);
+                            } finally {
+                                assertThat(buffer.isRecycled()).isTrue();
+                            }
+                        })
+                .hasCauseInstanceOf(TestException.class);
```

Comment:
```suggestion
    void testBuffersRecycledOnError() throws IOException {
        NetworkBuffer buffer = getBuffer();
        try (ChannelStateWriterImpl writer =
                new ChannelStateWriterImpl(
                        TASK_NAME, new ConcurrentHashMap<>(), failingWorker(), 5)) {
            writer.open();
            assertThatThrownBy(() -> callAddInputData(writer, buffer))
                    .isInstanceOf(RuntimeException.class)
                    .hasCauseInstanceOf(TestException.class);
            assertThat(buffer.isRecycled()).isTrue();
        }
```
We can assert on the exception here in a more-specific manner as well. This also improves the readability of the test.